pkg/outputs/async_kafka/async_kafka.go

```diff
@@ -189,7 +189,13 @@ func (output *AsyncKafka) Execute(workerIndex int, msgs []*core.Msg) error {
 			return errors.Annotatef(err, "topic: %v", topic)
 		}
 
-		partition := workerIndex % len(partitions)
+		// data with the same primary key goes to the same partition,
+		// if there is no primary key, use unique index;
+		// if there is no primary key and unique index, use partition 0.
```

Comment:
if there is no primary key and unique index, use random partition should be used.